config/builtin_locks.go

```diff
@@ -0,0 +1,26 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/gruntwork-io/terragrunt/locks"
+	"github.com/gruntwork-io/terragrunt/locks/dynamodb"
+)
+
+// lockFactory provides an implementation of Lock with the provided configuration map
+type lockFactory func(map[string]string) (locks.Lock, error)
+
+// lookupLock returns the implementation for the named lock or returns an error if
+// it is not found
+func lookupLock(name string, conf map[string]string) (locks.Lock, error) {
+	f, ok := builtinLocks[name]
+	if !ok {
+		return nil, fmt.Errorf("no Lock implementation found for %s", name)
```

Comment:
To make it easier to check error cases with automated tests, and to allow clients of the code to make better decisions on how to handle errors, it's a good idea to define custom error types. In this case, a `LockNotFound` type (just a type alias for a `string` with an `Error` method) would make this error easier to handle programmatically. Also, we try to wrap each error with a call to `errors.WithStacktrace` so we can trace down the cause more easily. See https://github.com/gruntwork-io/terragrunt#error-handling for more details.